AudiusProject/audius-client
# [C-1184] Fetch lineups after coming back online

### Description
Problem: 
Lineups are not loading when returning to online after the app started offline.

Solution: 
This fix switches `setupBackend` to a best-effort approach with retrying each time we return to online.

### Dragons

Changing the init logic for `setupBackend` could lead us to bad race conditions or failed requests. We may need to add `waitForBackendSetup` to more of our fetch sagas to ensure they only run after the `apiClient` is ready. I'd welcome guidance here from someone who knows the setup logic better.

### How Has This Been Tested?

Simulator and device
1. Launch app in airplane mode
2. Observe offline indicator
3. Turn off airplane mode
4. Optionally navigate in app while offline
5. Content on current page should load as normal
6. Navigate around after, content should continue to load normally

### How will this change be monitored?

Keep an eye on error counts in test and prod. See if it's causing an increase in random crashes/logged errors
